src/dune_rpc_impl/server.ml

```diff
@@ -22,12 +22,21 @@ let diagnostic_of_error : Build_system.Error.t -> Dune_rpc_private.Diagnostic.t
   let loc = message.loc in
   let message = Pp.map_tags (Pp.concat message.paragraphs) ~f:(fun _ -> ()) in
   let id = Build_system.Error.id m |> Diagnostic.Id.create in
+  let promotion =
+    match Build_system.Error.promotion m with
+    | None -> []
+    | Some { in_source; in_build } ->
+      [ { Diagnostic.Promotion.in_source = Path.Source.to_string in_source
```

Comment:
It seems a little strange to expose this as a list but only ever report singleton lists. Do we need to make a change to `cram` or a runtest alias to report these?

Reply:
The idea is to support multiple promotions in one action in the future. Users have been trying to do this meanwhile:

```
(progn
  (diff ..)
  (diff ..))
```

But it doesn't quite work as expected because if the first promotion fails, the second will not be offered.